docs/src/tutorials/pbt-and-tla.md

```diff
@@ -0,0 +1,1001 @@
+Choosing one determines \\(spender_L\\) (satisfying \\(c_2\\) automatically), and \\(sender_T\\) by \\(c_5\\). 
+What is left is selecting \\(to\\), which has \\(N_{addr}-2\\) candidates, by \\(c_3\\), for a total of \\(N_{addr}(N_{addr} - 1)( N_{addr} -2)\\) combinations, which satisfy all constraints.
+
+In summary, the probability of randomly selecting addresses satisfying \\(c_1,\dots,c_5\\) is 
+\\[
+P(c_1,\dots, c_5) = \frac{N_{addr}(N_{addr} - 1)(N_{addr} - 2)}{N_{addr}^7} = \frac{(N_{addr} - 1)(N_{addr} - 2)}{N_{addr}^6}
+\\]
+
+To determine \\(P(c_6)\\), we observe that there are \\(N_{val}^3\\) total possible selections of triples \\((v_L,v_T,v_H)\\). 
+The number of triples that satisfies \\(c_6\\) is
+\\[
+\sum_{a=1}^{N_{val}} \sum_{b=1}^{N_{val}} \sum_{c=1}^{N_{val}} \textbf{1}_{c \ge b > a > 1}
+\\]
+
+Where \\(\textbf{1}_{c \ge b > a > 1}\\) is the [indicator function](https://en.wikipedia.org/wiki/Indicator_function) of the event \\(c \ge b > a > 1\\), that is, equal to \\(1\\) iff \\(c \ge b > a > 1\\) and 0 otherwise.
+We can therefore simplify the above sum to 
+
+\\[
+\sum_{a=2}^{N_{val}} \sum_{b=a+1}^{N_{val}} \sum_{c=b}^{N_{val}} 1
+\\]
+
+which has a closed-form solution \\(\frac{N_{val}(N_{val}^2 - 3N_{val} + 2)}{6}\\), giving us the following probability of choosing values satisfying \\(c_6\\):
+
+\\[
+P(c_6) = \frac{N_{val}(N_{val}^2 - 3N_{val} + 2)}{6N_{val}^3} = \frac{N_{val}^2 - 3N_{val} + 2}{6N_{val}^2}
+\\]
+
+Combining these results gives us:
+
+\\[
+P(\omega \mid \alpha) = P(c_1,\dots,c_5)\cdot P(c_6) = \frac{(N_{addr} - 1)(N_{addr} - 2)(N_{val}^2 - 3N_{val} + 2)}{6N_{val}^2N_{addr}^6}
+\\]
+
+### 8.3 Probability of finding an invariant violation
+ 
+Recall the conditional probability rule
+\\[
+P(\omega) =  P(\alpha \cap \omega) = P(\omega \mid \alpha) \cdot P(\alpha)
+\\]
+
+We can instantiate it with the computed probabilities, to determine:
+
+\\[
+P(\omega) = \frac{12(N_{addr} - 1)(N_{addr} - 2)(N_{val}^2 - 3N_{val} + 2)}{6^6N_{val}^2N_{addr}^6} = \frac{(N_{addr} - 1)(N_{addr} - 2)(N_{val}^2 - 3N_{val} + 2)}{3888 N_{val}^2N_{addr}^6}
+\\]
+
+which equals \\(\frac{19}{31492800}\\) (approximately \\(6 \cdot 10^{-7}\\)) for \\((N_{val},N_{addr}) = (20,3)\\).
+
+Interestingly, the probability is asymptotically constant w.r.t. \\(N_{val}\\), but it is asymptotically inversely proportional to \\(N_{addr}^4\\).
+
+We can see that for the following increases in address space size, the probability of finding an invariant by random exploration drops drastically:
+Let us fix \\(N_{val} = 20\\) and increase \\(N_{addr}\\):
+  - \\(N_{addr} = 10\\): \\(P(\omega) \doteq 1.5 \cdot 10^{-8}\\)
+  - \\(N_{addr} = 50\\): \\(P(\omega) \doteq 3.3 \cdot 10^{-11}\\)
+  - \\(N_{addr} = 100\\): \\(P(\omega) \doteq 2.1 \cdot 10^{-12}\\)
+  - \\(N_{addr} = 1000\\): \\(P(\omega) \doteq 2.2 \cdot 10^{-16}\\)
+
+For reference, the space of 20byte address admits \\(2^{160}\\) unique values.
```

Comment:
I don't understand this last sentence, or how it's related to the above.

Reply:
Okay, then maybe that's what the sentence should say?
Though I'm not sure it's needed, since we're following the small-model hypothesis throughout the doc?